staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/mutating/dispatcher.go

```diff
@@ -133,7 +133,28 @@ func (a *mutatingDispatcher) Dispatch(ctx context.Context, attr admission.Attrib
 			round = 1
 		}
 		changed, err := a.callAttrMutatingHook(ctx, hook, invocation, versionedAttr, o, round, i)
-		admissionmetrics.Metrics.ObserveWebhook(time.Since(t), err != nil, versionedAttr.Attributes, "admit", hook.Name)
+		ignoreClientCallFailures := hook.FailurePolicy != nil && *hook.FailurePolicy == v1beta1.Ignore
+		rejected := false
+		if err != nil {
+			// ErrCallingWebhook is ignored if the webhook is configured to failopen.
+			// Otherwise the request is rejected.
+			if _, ok := err.(*webhookutil.ErrCallingWebhook); !ok || !ignoreClientCallFailures {
+				rejected = true
+				var errorType admissionmetrics.WebhookRejectionErrorType
+				rejectionCode := 0
+				switch err.(type) {
```

Comment:
this is fairly complex to follow, can we simplify to a top-level switch?

```go
if err != nil {
  switch err := err.(type) {

    case *webhookutil.ErrCallingWebhook:
      if !ignoreClientCallFailures {
        rejected = true
        admissionmetrics.Metrics.ObserveWebhookRejection(hook.Name, "admit", string(versionedAttr.Attributes.GetOperation()), 0, admissionmetrics.WebhookRejectionCallingWebhookError)
      }

    case *webhookutil.ErrWebhookRejection:
      rejected = true
      admissionmetrics.Metrics.ObserveWebhookRejection(hook.Name, "admit", string(versionedAttr.Attributes.GetOperation()), err.Status.ErrStatus.Code, admissionmetrics.WebhookRejectionNoError)

    default:
      rejected = true
      admissionmetrics.Metrics.ObserveWebhookRejection(hook.Name, "admit", string(versionedAttr.Attributes.GetOperation()), 0, admissionmetrics.WebhookRejectionAPIServerInternalError)
  }
}

admissionmetrics.Metrics.ObserveWebhook(time.Since(t), rejected, versionedAttr.Attributes, "admit", hook.Name)
```

Reply:
same comment applies to validation hook